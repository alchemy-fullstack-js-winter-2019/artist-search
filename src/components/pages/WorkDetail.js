import React, { PureComponent } from 'react';
import { getWorkDetail } from '../services/artistApi';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class ArtistDetail extends PureComponent {
  state = {
    lyrics: ''
  }
  static propTypes = {
    match: PropTypes.object
  }

  componentDidMount() {
    const { name, title } = this.props.match.params;
    getWorkDetail(name, title)
      .then(res => {
        this.setState({ lyrics: res.results });
      });
  }


  render() {
    
    return (
      <ul>
        { this.state.lyrics }
      </ul>
    );
  }
}
