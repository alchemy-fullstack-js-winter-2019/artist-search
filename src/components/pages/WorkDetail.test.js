import React from 'react';
import { shallow } from 'enzyme';
import WorkDetail from './WorkDetail';

jest.mock('../services/artistApi');

describe('Work Details', () => {
  it('renders work lyrics', () => {
    const wrapper = shallow(<WorkDetail 
      match={{ params: { name: 'Bob', title: 'Bob and his Greatest Hits' } }}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
