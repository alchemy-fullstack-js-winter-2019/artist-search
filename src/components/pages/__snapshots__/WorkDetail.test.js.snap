// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work Details renders work lyrics 1`] = `
<Fragment>
  <h2>
    Bob and his Greatest Hits
    :
  </h2>
</Fragment>
`;
